c_glib/arrow-glib/array.cpp

```diff
@@ -344,9 +344,40 @@ garrow_null_array_new(gint64 length)
 }
 
 
+G_DEFINE_TYPE(GArrowPrimitiveArray,             \
+              garrow_primitive_array,           \
+              GARROW_TYPE_ARRAY)
+
+static void
+garrow_primitive_array_init(GArrowPrimitiveArray *object)
+{
+}
+
+static void
+garrow_primitive_array_class_init(GArrowPrimitiveArrayClass *klass)
+{
+}
+
+/**
+ * garrow_primitive_array_get_buffer:
+ * @array: A #GArrowPrimitiveArray.
+ *
+ * Returns: (transfer full): The data of the array as #GArrowBuffer.
+ */
+GArrowBuffer *
+garrow_primitive_array_get_buffer(GArrowPrimitiveArray *array)
+{
+  auto arrow_array = garrow_array_get_raw(GARROW_ARRAY(array));
+  auto arrow_primitive_array =
+    static_cast<arrow::PrimitiveArray *>(arrow_array.get());
+  auto arrow_data = arrow_primitive_array->data();
+  return garrow_buffer_new_raw(&arrow_data);
```

Comment:
Here: https://github.com/apache/arrow/blob/master/c_glib/arrow-glib/buffer.cpp#L56

`garrow_buffer_finalize()` is called when `GArrowBuffer`'s reference count is 0.